apps/mark-scan/backend/src/util/controllerd.ts

```diff
@@ -0,0 +1,9 @@
+import { readFile } from 'fs/promises';
+
+const VIRTUAL_DEVICE_NAME = 'Accessible Controller Daemon Virtual Device';
+
+export async function isAccessibleControllerDaemonRunning(): Promise<boolean> {
+  return (await readFile('/proc/bus/input/devices')).includes(
+    VIRTUAL_DEVICE_NAME
+  );
```

Comment:
On my dev machine, virtual devices are cleaned up if the process crashes. So it does work. But I agree it would be better to do the other way. You just have to make sure you aren't catching the `grep controller` as a false positive. Are you trying this on hardware anytime soon?

Reply:
47294e08a

Reply:
I'll test this today and let you know what I find

Reply:
I think it might be better to treat this like a web service where you have a `/api/status` endpoint. Maybe create a Unix pipe that you can ping and get an expected response.

Alternatively, at least get the full path of the binary and verify that there is a process with that path? Longer term when it's run as a service we should be able to use the `service` command to check status.

Reply:
@kshen0 see my commit for a more terse way to to do it (per ChatGPT's suggestion). 

@eventualbuddha I agree it would be better to have stronger signal from the process, rather than treating the process running as a signal in itself, but what I have now seems good enough AFAICT and I'd like to get this merged because there's a bunch of other stuff here I want to get on `main`.